lib/rubygems/security.rb

```diff
@@ -337,17 +337,14 @@ class Exception < Gem::Exception; end
   DIGEST_NAME = 'SHA256' # :nodoc:
 
   ##
-  # Algorithm for creating the key pair used to sign gems
+  # Length of keys created by KEY_ALGORITHM
 
-  KEY_ALGORITHM =
-    if defined?(OpenSSL::PKey::RSA)
-      OpenSSL::PKey::RSA
-    end
+  KEY_LENGTH = 3072
 
   ##
-  # Length of keys created by KEY_ALGORITHM
+  # Key used for Elliptic Curve
 
-  KEY_LENGTH = 3072
+  EC_KEY = 'secp384r1'
```

Comment:
Any reasons for choosing this particular curve as the default? Also, thoughts on exposing a parameter to specify a different curve?